ehsm_kms_service/test/security_test/createkey.py

```diff
@@ -0,0 +1,217 @@
+
+#supported_keyspec = ["EH_AES_GCM_128", "EH_RSA_3072", ""]
+#origin = "EH_INTERNAL_KEY"
+#appid='1644220551486'
+#appkey = '1644220551486'
+common_invalid_params = ["123", "000", "aaaaaa", "**.,][][", "!=1 or", "12.13"]
+
+#class key_params:
+#    test_url = ""
+
+def init_params(appid, appkey, timestamp_id, keyspec, origin, sign=None):
+    payload = OrderedDict()
+    payload["keyspec"] = keyspec
+    payload["origin"] = origin
+    params = OrderedDict()
+    params["appid"] = appid
+    params["payload"] = urllib.parse.unquote(urllib.parse.urlencode(payload))
+    if timestamp_id in [0, 1, 2]:
+        params["timestamp"] = str(int((time.time()-540+(int(timestamp_id)*540)) * 1000))
+    else:
+        params["timestamp"] = timestamp_id
+    #params["timestamp"] = str(int((time.time()) * 1000))
+
+    sign_string = urllib.parse.unquote(urllib.parse.urlencode(params))
+    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), sign_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8').upper()
+
+    params["payload"] = payload
+    params["sign"] = sign
+    return params
+
+def init_params_with_invalid_sign(appid, appkey, timestamp_id, keyspec, origin, sign=None):
+    payload = OrderedDict()
+    payload["keyspec"] = keyspec
+    payload["origin"] = origin
+    params = OrderedDict()
+    params["appid"] = appid
+    params["payload"] = urllib.parse.unquote(urllib.parse.urlencode(payload))
+    params["timestamp"] = timestamp_id
+    #params["timestamp"] = str(int((time.time()) * 1000))
+
+    sign_string = urllib.parse.unquote(urllib.parse.urlencode(params))
+    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), sign_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8').upper()
+
+    params["timestamp"] = str(int(timestamp_id)+1)
+    params["payload"] = payload
+    params["sign"] = sign
+    return params
+
+
+@pytest.fixture(params=["e2014b79-ad08-49c4-897d-cbf39a2cdcce"])
+def appid_params(request):
+    return request.param
+
+@pytest.fixture(params=["123", "000", "aaaaaa", "**.,][][", "!=1 or", "12.13", "", "<script>alert(\"1111111\");</script>"])
+def invalid_appid_params(request):
+    return request.param
+
+
+@pytest.fixture(params=["D0G26swLnEAyCParTgKWqzZ5up6t5tmb"])
+def appkey_params(request):
+    return request.param
+
+@pytest.fixture(params=["123", "000", "aaaaaa", "**.,][][", "!=1 or", "12.13", "", "<script>alert(\"1111111\");</script>"])
+def invalid_appkey_params(request):
+    return request.param
+
+
+@pytest.fixture(params=[0, 1, 2])
+def timestamp_id_params(request):
+    return request.param
+
+@pytest.fixture(params=[str(int(time.time() * 1000)), str(int((time.time()+540) * 1000)), str(int((time.time()-540) * 1000))])
+def timestamp_params(request):
+    return request.param
+
+@pytest.fixture(params=["123", "000", "aaaaaa", "**.,][][", "!=1 or", "12.13", "<script>alert(\"1111111\");</script>", str(int((time.time()+1320) * 1000)), str(int((time.time()-1320) * 1000))])
```

Comment:
an injection like ' OR '1'='1 is for SQL.
The backend DB is CouchDB which is nonSQL.
Try to search if any noSQL injection samples based on javascript.